base/uk.ac.stfc.isis.ibex.ui.synoptic.editor/src/uk/ac/stfc/isis/ibex/ui/synoptic/editor/target/selector/TargetSelectorViewModel.java

```diff
@@ -0,0 +1,198 @@
+import uk.ac.stfc.isis.ibex.ui.synoptic.editor.model.UpdateTypes;
+
+/**
+ * The view model for the target selector.
+ */
+public class TargetSelectorViewModel extends ModelObject {
+    
+    private final SynopticViewModel synopticViewModel;
+    private boolean enabled;
+    private String name;
+    private String opi;
+    private int icon;
+    private String description;
+    
+    private static final String NONE_OPI = "NONE";
+    private final List<String> componentTypesList = ComponentType.componentTypeAlphaList();
+    
+    /**
+     * An array of component types for populating the drop down icon selector in the synoptic editor.
+     */
+    public final String[] componentTypesArray = componentTypesList.toArray(new String[0]);
+    
+    /**
+     * View model for the target selector panel.
+     * @param synopticViewModel the main synoptic view model
+     */
+    public TargetSelectorViewModel(final SynopticViewModel synopticViewModel) {
+        this.synopticViewModel = synopticViewModel;
+        addEnablementListener();
+    }
+    
+    private void addEnablementListener() {
+        synopticViewModel.addPropertyChangeListener("selectedComponents", new PropertyChangeListener() {
+            @Override
+            public void propertyChange(PropertyChangeEvent evt) {
+                if (synopticViewModel.getSingleSelectedComp() != null) {
+                    setEnabled(true);
+                    setName(synopticViewModel.getSingleSelectedComp().getName());
+                    setIconSelectionIndex(componentTypesList.indexOf(synopticViewModel.getSingleSelectedComp().type().name()));
+                    setOpi(synopticViewModel.getSingleSelectedComp().target().name());
+                } else {
+                    setName("");
+                    setIconSelectionIndex(0);
+                    setOpi(NONE_OPI);
+                    setEnabled(false);
+                }
+            }
+        });
+    }
+    
+    /**
+     * Whether the panel is enabled.
+     * @return true if enabled, false otherwise
+     */
+    public boolean isEnabled() {
+        return enabled;
+    }
+    
+    /**
+     * Sets whether the panel should be enabled.
+     * @param enabled true to enable, false to disable
+     */
+    public void setEnabled(boolean enabled) {
+        firePropertyChange("enabled", this.enabled, this.enabled = enabled);
+    }
+    
+    /**
+     * Gets the name of this component.
+     * @return the name of this component
+     */
+    public String getName() {
+        return name;
+    }
+    
+    /**
+     * Set the name of this component.
+     * @param name the new name
+     */
+    public void setName(String name) {
+        if (synopticViewModel.getSingleSelectedComp() == null) {
+            return;
+        }
+        firePropertyChange("name", this.name, this.name = name);
+        synopticViewModel.getSingleSelectedComp().setName(name);
```

Comment:
As far as I can tell the the selected component is either null, or where the name argument is coming from in the first place. Is this line redundant? Same question for equivalent lines in setOpi() and setIconSelectionIndex()

Reply:
This method is used in databinding from the UI classes so I don't think this is redundant (unless I've misinterpreted your question)? It needs to propagate the change from this view model to the synoptic view model so that the component tree updates with the correct names/icons.

Reply:
Oh yeah, you're right. Sorry I was being dumb, looking at the wrong thing